File: LeetCode/2125_NumberOfLaserBeamsInABank/lc2125.cpp
#include <algorithm>
#include <cassert>
#include <string>
#include <vector>

class Solution {
public:
  int numberOfBeams(std::vector<std::string>& bank) {
    int last = -1;
    int ans = 0;
    for (const std::string& row: bank) {
      const int count = std::ranges::count(row, '1');
      if (count > 0) {
        if (last == -1) {
          last = count;
        }
        else {
          ans += last * count;
          last = count;
        }
      }
    }
    return ans;
  }
};

int main() {
  Solution s;
  {
    std::vector<std::string> bank = {"011001", "000000", "010100", "001000"};
    assert(s.numberOfBeams(bank) == 8);
  }
  {
    std::vector<std::string> bank = {"000", "111", "000"};
    assert(s.numberOfBeams(bank) == 0);
  }
};

File: LeetCode/2125/golang/lc2125.go
package lc2125

import "strings"

// Runtime complexity: O(m*n); where m and n are the dimensions of `bank` 2D matrix.
// Auxiliary space: O(1).
// Subjective level: easy.
func numberOfBeams(bank []string) int {
	prev := 0
	ans := 0
	for _, row := range bank {
		curr := strings.Count(row, "1")
		if curr > 0 {
			ans += (curr * prev)
			prev = curr
		}
	}
	return ans
}
